src/nimony/sem.nim

```diff
@@ -1745,27 +1777,30 @@ proc findObjFieldAux(t: Cursor; name: StrId; level = 0): ObjField =
   else:
     if baseType.typeKind in {RefT, PtrT}:
       inc baseType
-    if baseType.typeKind == InvokeT:
-      inc baseType # get to root symbol
+    var baseInvokeArgs = skipInvoke(baseType)
     if baseType.kind == Symbol:
       let decl = getTypeSection(baseType.symId)
-      var objType = decl.body
-      # emulate objtypeImpl
-      if objType.typeKind in {RefT, PtrT}:
-        inc objType
-      result = findObjFieldAux(objType, name, level+1)
+      var newBindingBuf = default(TokenBuf)
+      var newBindings = initTable[SymId, Cursor]()
+      if baseInvokeArgs != default(Cursor):
+        newBindingBuf = createTokenBuf(16)
+        var sc = SubsContext(params: addr bindings)
+        while baseInvokeArgs.kind != ParRi:
+          subs(c, newBindingBuf, sc, baseInvokeArgs)
+          skip baseInvokeArgs
+        newBindingBuf.addParRi()
+        buildInvokeArgs(decl, newBindings, beginRead(newBindingBuf))
```

Comment:
Move to a helper proc and while you're at it add a long comment explaining why all these shenigans are necessary.

Reply:
Hopefully better now